bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/AbstractComponent.java

```diff
@@ -62,6 +64,7 @@
  */
 @NonNullByDefault
 public abstract class AbstractComponent<C extends AbstractChannelConfiguration> {
+    public static final String JSON_ATTRIBUTES_CHANNEL_ID = "json-attributes";
```

Comment:
Now this is added, it can be removed from all components.

As this channel exposes the json it is some sort of raw data channel. Something we try to prevend. All data should be modelled into the openHAB domain model (Things, Channels). Is there a particular reason or use-case where you want this to be exposed in a channel?

Reply:
The documentation states "The MQTT topic subscribed to receive a JSON dictionary payload and then set as sensor attributes." A full example is at https://www.home-assistant.io/integrations/sensor.mqtt/#json-attributes-topic-configuration. But basically, any component is allowed to stuff any information it wants into the extra "JSON Attributes". Its only structure is that it's JSON. AFAICT it's even allowed to add or remove attributes from that object at any time. And there's no metadata anywhere that describes what it might contain. It's really up to the device to throw anything it wants in there. So while I agree that I don't love a channel stuffed with JSON, I don't really know what else to do. It seems weird to be adding and removing channels as data is received/not received, but maybe that _is_ the best solution? Or do we just not do JSON Attributes at all, and if someone wants to have access to them, they have to manually configure a generic MQTT channel? If that were that case, it might be nice to be able to add manually configured channels to an existing Home Assistant MQTT Thing, rather than having to create a separate thing. Or possibly a hybrid approach - manually configured channels, but only necessary to state the attribute name in the JSON?

ZWave-JS-UI populates the JSON attributes topic, but I've never looked to see what useful information might be in there. Zigbee2MQTT has an upcoming change that button press events will be delivered with the new `Event` component, rather than the `DeviceTrigger` component, due to the latter having legacy issues not properly separating the button from the type of event. With the new `Event` implementation, the details of which button will be delivered in the JSON attributes. But the JSON Attributes on the `Event` component should be a Trigger channel anyway.

I'm fine with _not_ adding json-attributes to any new components in 4.3, keeping it where it was for back-compat for now, and then doing a separate PR for a different implementation if you think one of the above (or something completely different!) would be better.

Reply:
Per-component `JSON_ATTRIBUTES_CHANNEL_ID` has been removed with latest commits; longer term question of if we even want to expose JSON Attributes still pending.

Reply:
The binding _can_ rely on the json, and use it. Exposing the json by a channel to the end-user is something else. From what i understand now is that this JSON can have usefull (device specific) data  and the binding has no (structured) way of transforming it to channels.
The base functionality of the component is already defined and available in the topics. So the exposed data in this JSON is extra, still usefull but not mandatory. Due to the nature of this binding and how this all works i would propose to have this channel available as advanced. This would give a clear signal to users to not rely too much on it, or use it on own risk.
Have not checked, but i hope all channels are only updated when they are actually linked. As all moving all this JSON around can have a performance hit.

Reply:
Good call. Marked it as advanced.

NOT subscribing (on the MQTT side) to topics for channels that aren't linked is in the back-of-my-mind todo list. Cause yeah.... I've got a LOT of topics, and lots of things that I don't have linked.